stetho/src/main/java/com/facebook/stetho/inspector/network/NetworkEventReporterImpl.java

```diff
@@ -152,10 +152,45 @@ public void responseHeadersReceived(InspectorResponse response) {
           getResourceTypeHelper().determineResourceType(contentType) :
           Page.ResourceType.OTHER;
       receivedParams.response = responseJSON;
+      initAsyncPrettyPrinterForResponse(response, peerManager, receivedParams);
       peerManager.sendNotificationToPeers("Network.responseReceived", receivedParams);
     }
   }
 
+  static void initAsyncPrettyPrinterForResponse(
+      InspectorResponse response,
+      NetworkPeerManager peerManager,
+      Network.ResponseReceivedParams receivedParams) {
+    AsyncPrettyPrinterRegistry registry = peerManager.getAsyncPrettyPrinterRegistry();
+    AsyncPrettyPrinter asyncPrettyPrinter = createPrettyPrinterForResponse(
+        response,
+        registry,
+        receivedParams);
+    if (asyncPrettyPrinter != null) {
+      peerManager.getResponseBodyFileManager().associateAsyncPrettyPrinterWithId(
+          response.requestId(),
+          asyncPrettyPrinter);
+    }
+  }
+
+  @Nullable
+  private static AsyncPrettyPrinter createPrettyPrinterForResponse(
+      InspectorResponse response,
+      AsyncPrettyPrinterRegistry registry,
+      Network.ResponseReceivedParams receivedParams) {
+    for (int i = 0, count = response.headerCount(); i < count; i++) {
+      AsyncPrettyPrinterFactory factory = registry.lookup(response.headerName(i));
+      if (factory != null) {
+        receivedParams.type = factory.getPrettifiedType();
```

Comment:
This is ugly collusion.  Prefer not to mix unrelated concepts if you don't have to.  In this case, you can divide the method into `findAsyncPrettyPrinterFactory(...)` and then another function that calls `getInstance`.  By dividing functions into further granular logical components you can better express the logic you're after without uber functions that take large numbers ofa rguments (trust me, that's where you were headed *grin*)